app/controllers/assignment_swap_controller.rb

```diff
@@ -0,0 +1,34 @@
+class AssignmentSwapController < SecureController
+
+  skip_before_action :authenticate_user!
```

Comment:
This shouldn't be a `SecureController` and then you don't need to skip the action.